File: src/main/java/com/hris/HRIS/model/PayrollReportModel.java
package com.hris.HRIS.model;

import com.hris.HRIS.dto.PayrollReportItem;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@Getter
@Setter
@ToString

@Document(collection = "payrollreport")
public class PayrollReportModel {
    @Id
    private String id;
    private String organizationId;
    private String email;
    private String reportDescription;
    private String reportType;
    private Integer workedDays;
    private List<PayrollReportItem> payItems;
    private List<PayrollReportItem> deductions;
    private double totalEarnings;
    private double totalDeductions;
    private double netPay;
    private String payPeriod; // July 2024, etc...
    private String reportGeneratedDate;
    private String status;
}

File: client/src/app/shared/data-models/payroll-report.model.ts
export interface PayrollReportModel{
    id: string;
    organizationId: string;
    email: string;
    reportDescription: string;
    reportType: string;
    workedDays: number;
    payItems: { itemName: string, description: string, amount: number }[];
    deductions: { itemName: string, description: string, amount: number }[];
    totalEarnings: number;
    totalDeductions: number;
    netPay: number;
    payPeriod: string; // July 2024, etc...
    reportGeneratedDate: string;
    status: string;
}